hail/src/main/scala/is/hail/expr/types/physical/PNDArray.scala

```diff
@@ -36,7 +36,7 @@ final case class PNDArray(elementType: PType, nDims: Int, override val required:
       .loadField(shape, idx))
 
     if (nDims == 0) {
-      0L
```

Comment:
was this uncovered by removing an optimizer rule?  If not, we should add a test for this case.

Reply:
There are tests for this. This is for 0 dimensional ndarrays, like 

`hl._ndarray(5)` or `np.array(5)`. They have 0 dimensions and their shape is the empty tuple.

Reply:
What you're telling me is that:
```
Array.range(0, nDims).foldLeft(const(1L)) {...}
```
Will do exactly the right thing then.

Reply:
Yup! Good catch. It was a separate case when I didn't understand the need to support 0 dimensional NDArrays, and I didn't realize the condition could be removed now that we acknowledge 0 dimensional NDArrays as having 1 element.